src/util/getTileInfo.ts

```diff
@@ -81,36 +87,50 @@ function getStairs(tiles: TileType[][], x: number, y: number) {
   const type = getTile(tiles, x, y);
   const topType = getTile(tiles, x, y, "top");
   const leftType = getTile(tiles, x, y, "left");
+  const rightType = getTile(tiles, x, y, "right");
+  const topLeftType = getTile(tiles, x, y, "topLeft");
+  const bottomLeftType = getTile(tiles, x, y, "bottomLeft");
+  const topRightType = getTile(tiles, x, y, "topRight");
+  
+
 
   if (isTile(topType) && isTile(type)) {
     const diff = Number(topType) - Number(type);
 
     if (diff === 1) {
-      const destinationStairs = getStairs(
-        tiles,
-        x + offsets.top.x,
-        y + offsets.top.y
-      );
-
-      if (destinationStairs == null || destinationStairs.direction === 0) {
```

Comment:
The problem with your screen is from this check. Without this as u can see all seem okay.

![fix_stairs](https://user-images.githubusercontent.com/2702557/104208818-3c7eb680-5431-11eb-85f2-c5bcdf591501.PNG)

![fix_stairs2](https://user-images.githubusercontent.com/2702557/104208943-60da9300-5431-11eb-8d28-d4cbb58b78b3.PNG)